Update docs about preview from pull/merge requests (#7823)
readthedocs/readthedocs.org: docs/pull-requests.rst

> Preview Documentation from Pull Requests

Reviewer: We should probably just call these Pull Requests in the content here to match the title, instead of `Pull/Merge requests`. I think everyone knows what PR's are at this point.

Author: +1 on that

========================================

> Read the Docs allows to build and preview your documentation from pull/merge requests.

Reviewer: ```suggestion
Read the Docs allows you to build and preview your documentation from pull/merge requests.
```

> You can enable it by: 

Reviewer: ```suggestion
To enable this feature: 
```

#. Go to your project dashboard
#. Go to :guilabel:`Admin > Advanced settings`
#. Enable the :guilabel:`Build pull requests for this project` option
#. Click on :guilabel:`Save`

Features
--------

- **Build on Pull/Merge Request Events:** We create and build a new version when a pull/merge request is open,
  and when a new commit has been pushed.

- **Warning Banner:** A warning banner is shown at the top of the documentation
  to let users know that this isn't the main documentation for the project.

  .. note:: This feature is available only for :doc:`Sphinx projects </intro/getting-started-with-sphinx>`.

> - **Build Status Report:** When a build is triggered, a build pending notification is send with a link to the build log,

Reviewer: This should be above the Warning banner, it's a much bigger feature. I think we can probably just remove the warning banner section, since it's not really that useful to document?

Author: I think we should document it somewhere, as people some times expect this kind of fetatures to be on mkdocs projects as well. https://github.com/readthedocs/readthedocs.org/issues/7810

> - **Build Status Report:** When a build is triggered, a build pending notification is send with a link to the build log,

Reviewer: ```suggestion
- **Build Status Report:** When a build is triggered, a build pending notification is sent with a link to the build log.
```

>   and when the build is done we send a success notification with the link to the preview or a failure notification with a link to the build log.

Reviewer: ```suggestion
  When the build finishes we send a success notification with the link to the preview or a failure notification with a link to the build log.
```

.. figure:: /_static/images/github-build-status-reporting.gif
   :align: center
   :alt: GitHub Build Status Reporting for Pull Requests.
   :figwidth: 80%
   :target: ../_static/images/guides/github-build-status-reporting.gif

   GitHub build status reporting

Privacy levels
--------------

.. note::

   Privacy levels are only supported on :doc:`/commercial/index`.

All docs built from a pull/merge requests are private by default.
Currently, this can't be changed, but we are planning to support this.

Limitations
-----------

- Auto-builds for pull/merge requests have

>   :doc:`the same limitations as regular documentation builds </builds>`.

Reviewer: What does this mean? We should clarify this, or put it at the end.

Author: I think this was meant to say about the build memory/time available, I can make it more clear.

- Only available for GitHub and GitLab.

> - Additional formats like PDF and Epub aren't build.

Reviewer: ```suggestion
- Additional formats like PDF and Epub aren't built to produce results quicker. 
```

- Searches will default to the default experience for your tool.
  This is a feature we plan to add,

>   but don't want to overwhelm our search indexes used in production.

Reviewer: Is this true? I thought search just didn't work :/

Author: It should just work, as they are regular versions, we can enable it by removing this line

https://github.com/readthedocs/readthedocs.org/blob/74df71db645222926659c265c33b9abf577a9972/readthedocs/projects/tasks.py#L1318-L1320

Author: now with the custom options for search I think this is useful, so users can test the results.

Reviewer: Yea, it will just destroy our search indexes. I don't want to break prod search by indexing PR's, so we probably want a different cluster for it, if nothing else.

- The built documentation is kept for 90 days after the pull/merge request has been closed or merged.

Troubleshooting
---------------

#. **Pull Requests builds are not triggering**.
   We only support GitHub and GitLab currently. You need to make sure
   that your Read the Docs account is connected with that providers social account.
   You can check this by going to your `profile settings`_.

> #. **Build status is not being reported on your Pull/Merge Request**.

Reviewer: ```suggestion
#. **Build status is not being reported to your VCS provider**.
```

   You need to make sure that you have granted access to the Read the Docs

>    `OAuth App`_ to your/organizations GitHub account. If you do not see "Read the Docs"

Reviewer: ```suggestion
   `OAuth App`_ to your personal or organization GitHub account. If you do not see "Read the Docs"
```

>    among the `OAuth App`_, you might need to disconnect and reconnect to GitHub service.

Reviewer: This links to GitHub, what about GitLab? 

```suggestion
   in the `OAuth App`_ settings, you might need to disconnect and reconnect to GitHub service.
```

Author: Looks like we don't have that kind of problems with gitlab, if you have access to a repo you post the status.

   Also make sure your webhook is properly setup

>    to handle events. You can setup or ``re-sync`` the webhook from your projects admin dashboard.

Reviewer: ```suggestion
   to handle events related to pull requests. You can setup or ``re-sync`` the webhook from your projects admin dashboard.
```

Preview Documentation from Pull Requests
========================================

Read the Docs allows you to build and preview your documentation from pull requests.
To enable this feature:

#. Go to your project dashboard
#. Go to :guilabel:`Admin > Advanced settings`
#. Enable the :guilabel:`Build pull requests for this project` option
#. Click on :guilabel:`Save`

Features
--------

- **Build on Pull Request Events:** We create and build a new version when a pull request is open,
  and when a new commit has been pushed.

- **Build Status Report:** When a build is triggered, a build pending notification is sent with a link to the build log.
  When the build finishes we send a success notification with the link to the preview or a failure notification with a link to the build log.

- **Warning Banner:** A warning banner is shown at the top of the documentation
  to let users know that this isn't the main documentation for the project.

>   .. note:: This feature is available only for :doc:`Sphinx projects </intro/getting-started-with-sphinx>`.

Reviewer: ```suggestion
  .. note:: Warning banners are available only for :doc:`Sphinx projects </intro/getting-started-with-sphinx>`.
```

Preview Documentation from Pull Requests
========================================

Read the Docs allows to build and preview your documentation from pull/merge requests.
You can enable it by: 

#. Go to your project dashboard
#. Go to :guilabel:`Admin > Advanced settings`
#. Enable the :guilabel:`Build pull requests for this project` option
#. Click on :guilabel:`Save`

Features
--------

- **Build on Pull/Merge Request Events:** We create and build a new version when a pull/merge request is open,
  and when a new commit has been pushed.

- **Warning Banner:** A warning banner is shown at the top of the documentation

>   to let users know that this isn't the main documentation for the project.

Author: Need to update this to mention this is sphinx only
